lte/gateway/python/magma/subscriberdb/store/sqlite.py

```diff
@@ -164,26 +217,35 @@ def resync(self, subscribers):
         Args:
             subscribers - list of subscribers to be in the store.
         """
-        with self.conn:
-            # Capture the current state of the subscribers
-            res = self.conn.execute("SELECT subscriber_id, data FROM subscriberdb")
-            current_state = {}
-            for row in res:
-                sub = SubscriberData()
-                sub.ParseFromString(row[1])
-                current_state[row[0]] = sub.state
-
-            # Clear all subscribers
-            self.conn.execute("DELETE FROM subscriberdb")
-
-            # Add the subscribers with the current state
-            for sub in subscribers:
-                sid = SIDUtils.to_str(sub.sid)
-                if sid in current_state:
-                    sub.state.CopyFrom(current_state[sid])
-                data_str = sub.SerializeToString()
-                self.conn.execute("INSERT INTO subscriberdb(subscriber_id, data) "
-                                  "VALUES (?, ?)", (sid, data_str))
+        bucket_subs = defaultdict(list)
+        for sub in subscribers:
+            sid = SIDUtils.to_str(sub.sid)
+            bucket_subs[int(sid[-SID_DIGITS:])].append(sid)
+
+        for i, db_location in enumerate(self._db_location):
+            conn = sqlite3.connect(db_location, uri=True)
+            try:
+                with conn:
+                    # Capture the current state of the subscribers
+                    res = conn.execute("SELECT subscriber_id, data FROM subscriberdb")
+                    current_state = {}
+                    for row in res:
+                        sub = SubscriberData()
+                        sub.ParseFromString(row[1])
+                        current_state[row[0]] = sub.state
+
+                    # Clear all subscribers
+                    conn.execute("DELETE FROM subscriberdb")
+
+                    # Add the subscribers with the current state
+                    for sid in bucket_subs[i]:
+                        if sid in current_state:
+                            sub.state.CopyFrom(current_state[sid])
+                            data_str = sub.SerializeToString()
+                            conn.execute("INSERT INTO subscriberdb(subscriber_id, data) "
+                                         "VALUES (?, ?)", (sid, data_str))
```

Comment:
is indentation correct here??
You are changing behavior

Reply:
good catch! indentation is wrong...

Reply:
on line 223 I should have stored the subs and on 241 items should be subs not sid, etc...